packages/eslint-plugin/rules/valid-sprintf.js

```diff
@@ -75,44 +105,64 @@ module.exports = {
 				}
 
 				if ( ! candidates.length ) {
-					context.report(
+					context.report( {
 						node,
-						'sprintf must be called with a valid format string'
-					);
+						messageId: 'invalidFormatString',
+					} );
 					return;
 				}
 
 				let numPlaceholders;
-				for ( let i = 0; i < candidates.length; i++ ) {
-					const match = candidates[ i ].match( REGEXP_PLACEHOLDER );
+				for ( const candidate of candidates ) {
+					const allMatches = candidate.match(
+						REGEXP_SPRINTF_PLACEHOLDER
+					);
+					const unorderedMatches = candidate.match(
+						REGEXP_SPRINTF_PLACEHOLDER_UNORDERED
+					);
```

Comment:
I don't know why [the `no-unused-vars-before-return` rule](https://github.com/WordPress/gutenberg/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-before-return.md) is not flagging it, but if this assigned value only becomes relevant after the first condition, I might suggest to move it closer to where it's first referenced. May have a bigger impact depending how likely it is that the execution would abort by the first `return`.